contrib/platform/test/com/sun/jna/platform/win32/NamedPipeTest.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+package com.sun.jna.platform.win32;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import org.junit.Test;
+
+import com.sun.jna.platform.win32.WinDef.ULONGByReference;
+import com.sun.jna.platform.win32.WinNT.HANDLE;
+import com.sun.jna.ptr.IntByReference;
+
+/**
+ * @author lgoldstein
+ */
+public class NamedPipeTest extends AbstractWin32TestSupport {
+    public NamedPipeTest() {
+        super();
+    }
+
+    @Test
+    public void testNamedPipeServerAPI() {
+        String    pipeName="\\\\.\\pipe\\" + getCurrentTestName();
+        HANDLE    hNamedPipe=Kernel32.INSTANCE.CreateNamedPipe(pipeName,
+                WinBase.PIPE_ACCESS_DUPLEX,        // dwOpenMode
+                WinBase.PIPE_TYPE_BYTE | WinBase.PIPE_READMODE_BYTE | WinBase.PIPE_WAIT,    // dwPipeMode
+                1,    // nMaxInstances,
+                Byte.MAX_VALUE,    // nOutBufferSize,
+                Byte.MAX_VALUE,    // nInBufferSize,
+                1000,    // nDefaultTimeOut,
+                null);    // lpSecurityAttributes
+        assertCallSucceeded("CreateNamedPipe", !WinBase.INVALID_HANDLE_VALUE.equals(hNamedPipe));
+
+        // NOTE: we don't really care what the returned values are only that the call succeeds
+        try {
+            IntByReference    lpFlags=new IntByReference(0);
+            IntByReference    lpOutBuffferSize=new IntByReference(0);
+            IntByReference    lpInBufferSize=new IntByReference(0);
+                IntByReference    lpMaxInstances=new IntByReference(0);
+                assertCallSucceeded("GetNamedPipeInfo",
+                                    Kernel32.INSTANCE.GetNamedPipeInfo(hNamedPipe, lpFlags,
+                                            lpOutBuffferSize, lpInBufferSize, lpMaxInstances));
+
+                ULONGByReference    ServerProcessId=new ULONGByReference();
+                assertCallSucceeded("GetNamedPipeServerProcessId", Kernel32.INSTANCE.GetNamedPipeServerProcessId(hNamedPipe, ServerProcessId));
+
+                ULONGByReference ServerSessionId=new ULONGByReference();
+                assertCallSucceeded("GetNamedPipeServerSessionId", Kernel32.INSTANCE.GetNamedPipeServerSessionId(hNamedPipe, ServerSessionId));
+
+                assertCallSucceeded("DisconnectNamedPipe", Kernel32.INSTANCE.DisconnectNamedPipe(hNamedPipe));
+        } finally {    // clean up
+            assertCallSucceeded("Named pipe handle close", Kernel32.INSTANCE.CloseHandle(hNamedPipe));
```

Comment:
Actually, it goes away after *all* the tests in the class have been executed, and more exactly it goes away when the _junit_ ANT task is finished. This means that if there are many such leaks (memory, handles, sockets, etc.) it may affect other tests. I believe that a test should attempt (within reason) to clean up after itself as much as possible - either via _try/finally_ or _setUp()/tearDown()_.